internal/storage/v1/elasticsearch/factory.go

```diff
@@ -65,38 +49,40 @@ type Factory struct {
 	client atomic.Pointer[es.Client]
 
 	pwdFileWatcher *fswatcher.FSWatcher
+
+	templateBuilder es.TemplateBuilder
 }
 
-// NewFactory creates a new Factory.
-func NewFactory() *Factory {
-	return &Factory{
+// NewFactoryBase creates a new FactoryBase.
+func NewFactoryBase() *FactoryBase {
```

Comment:
make these private. As I understand they are only needed for v1 factory because of legacy way of initializing. So next year when we clean up the v1 code we can also remove these private constructors.
```suggestion
func newPrimaryFactoryBase() *FactoryBase {
```